dotnet/targets/Xamarin.Shared.Sdk.targets

```
@@ -1182,13 +1182,13 @@
 	<Target Name="_ComputeDynamicLibrariesToReidentify">
 		<PropertyGroup>
 			<_ExecutablePathPrefix Condition="'$(_PlatformName)' == 'iOS' Or '$(_PlatformName)' == 'tvOS'">@executable_path/</_ExecutablePathPrefix>
-			<_ExecutablePathPrefix Condition="'$(_PlatformName)' == 'macOS' Or '$(_PlatformName)' == 'MacCatalyst'">@executable_path/../$(_CustomBundleName)/</_ExecutablePathPrefix>
+			<_ExecutablePathPrefix Condition="'$(_PlatformName)' == 'macOS' Or '$(_PlatformName)' == 'MacCatalyst'">@executable_path/../../</_ExecutablePathPrefix>
 		</PropertyGroup>
 		<ItemGroup>
 			<_DynamicLibraryToReidentify Include="@(_FileNativeReference)" Condition="'%(Extension)' == '.dylib'">
-				<DynamicLibraryId>$(_ExecutablePathPrefix)%(Filename)%(Extension)</DynamicLibraryId>
 				<SourceItemGroup>_FileNativeReference</SourceItemGroup>
-				<ReidentifiedPath>$(_IntermediateNativeLibraryDir)/%(Filename)%(Extension)</ReidentifiedPath>
+				<DynamicLibraryId>$([System.String]::Copy('$(_ExecutablePathPrefix)%(RelativePath)').Replace('\', '/'))</DynamicLibraryId>
```

It looks like this doesn't quite work, because not all items have `RelativePath`:

>  error MSB4096: The item "/Users/builder/azdo/_work/3/s/xamarin-macios/tests/xharness/tmp-test-dir/monotouch-test3898/bin/Debug/net7.0-macos/bindings-test.resources/libtest.xcframework/macos-arm64_x86_64/libtest.a" in item list "_FileNativeReference" does not define a value for metadata "RelativePath". In order to use this metadata, either qualify it by specifying %(_FileNativeReference.RelativePath), or ensure that all items in this list define a value for this metadata. [/Users/builder/azdo/_work/3/s/xamarin-macios/tests/xharness/tmp-test-dir/monotouch-test3684/monotouch-test.csproj]

---

On another note, maybe it would be a good idea to only set `DynamicLibraryId` and `ReidentifiedPath` if they're not already set, that way they can be overridden:

```suggestion
				<DynamicLibraryId Condition="'%(_FileNativeReference.DynamicLibraryId)' == ''">$([System.String]::Copy('$(_ExecutablePathPrefix)%(RelativePath)').Replace('\', '/'))</DynamicLibraryId>
```

---

I do not understand how this can happen since the condition `Condition="'%(Extension)' == '.dylib'"` should filter out from the list all static libraries and only keep dynamic libraries that do have `RelativePath`. How is libtest.a being processed with that condition?

---

Thanks for your help. I have updated the patch with all your suggestions